### PR TITLE
Update RustCli Parsing to process pkgId, and introduce manual override

### PR DESCRIPTION
**Pull request by FernandoRojo**
Allows users to manually override rustCLI detector by setting an environment variable DisableRustCliScan=True

Updates cargo metadata output processing to adhere to changes introduced that make the output follow the PkgId Spec: https://doc.rust-lang.org/cargo/reference/pkgid-spec.html

Changes to the metadata output were introduced in 1.77: https://github.com/rust-lang/cargo/pull/12914

### CODE REVIEW

File: src/Microsoft.ComponentDetection.Detectors/rust/RustCliDetector.cs [hunk 1]
```diff
@@ -187,9 +203,81 @@ private static bool ShouldFallbackFromError(string error)
         return true;
     }
 
-    private static bool ParseDependency(string dependency, out string packageName, out string version, out string source)
+    private static bool ParseDependencyMetadata(string dependency, out string packageName, out string version, out string source)
     {
-        var match = DependencyFormatRegex.Match(dependency);
+        // There are a few different formats for pkgids: https://doc.rust-lang.org/cargo/commands/cargo-pkgid.html#description
```
Comment (FernandoRojo) [inline src/Microsoft.ComponentDetection.Detectors/rust/RustCliDetector.cs:208]: There must be a better way to do this, was pressed for time but if there are any suggestions here I'm all ears

File: src/Microsoft.ComponentDetection.Detectors/rust/RustCliDetector.cs [hunk 2]
```diff
@@ -187,9 +203,81 @@ private static bool ShouldFallbackFromError(string error)
...
+
+            // if there is a third group, then the second must be packageName, third is version.
+            if (thirdGroup.Success)
+            {
+                packageName = secondGroup.Value;
+                version = thirdGroup.Value;
+            }
+
+            // if there is no third group, but there is a second, the second group could be either the name or the version, check if the value starts with a number (not allowed)
+            else if (secondGroup.Success)
```
Comment (cobya) [inline src/Microsoft.ComponentDetection.Detectors/rust/RustCliDetector.cs:254]: Can we try and get a test case covering these two below matches?

**Comment by cobya** [comment]
@FernandoRojo to verify, the new locations in snapshot verification are expected correct?

File: src/Microsoft.ComponentDetection.Detectors/rust/RustCliDetector.cs (hunk 2 above)
Comment (FernandoRojo) [reply src/Microsoft.ComponentDetection.Detectors/rust/RustCliDetector.cs]: changed the logic to use the pkgIds directly, so this whole parsing method is no longer needed.

**Review by cobya — APPROVED** [review]

**Pull request merged**